tools/linter/clang_tidy/max_tokens_pragma.py

```diff
@@ -0,0 +1,101 @@
+import argparse
+import re
+from typing import List
+
+
+DEFAULT_MAX_TOKEN_COUNT = 1
+MAX_TOKENS_CHECK_DIAG_NAME = "misc-max-tokens"
+MAX_TOKENS_PRAGMA_PATERN = r"#pragma\s+clang\s+max_tokens_total\s+(\d+)"
```

Comment:
```suggestion
MAX_TOKENS_PRAGMA_PATTERN = r"^#pragma\s+clang\s+max_tokens_total\s+(\d+)$"
```

the line markers might make you need some `textwrap.dedent`s in the tests below btw